### PR TITLE
[MOB-10033] Example App: inbox

### PR DESCRIPTION
**Pull request by lposen**
## 🔹 JIRA Ticket(s) if any

* [MOB-10033](https://iterable.atlassian.net/browse/MOB-10033)

## ✏️ Description

Added an inbox to the example app

## Testing

- cd into the root directory
- run `yarn install`
- cd into the example directory
- follow the instructions for setup in the example directories readme

### iOS
Simulator Screenshot - iPhone 11 - 2024-11-06 at 16 06 40

### Android
Screenshot 2024-11-06 at 4 06 57 PM

[MOB-10033]: https://iterable.atlassian.net/browse/MOB-10033?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/IterableInboxMessageList.tsx [hunk 1]
```diff
@@ -52,9 +52,7 @@ export const IterableInboxMessageList = ({
         swipingCheck={setSwiping}
         messageListItemLayout={messageListItemLayout}
         deleteRow={(messageId: string) => deleteRow(messageId)}
-        handleMessageSelect={(messageId: string, i: number) =>
-          handleMessageSelect(messageId, i)
-        }
+        handleMessageSelect={handleMessageSelect}
```
Comment (evantk91) [inline src/IterableInboxMessageList.tsx:55]: Whats the reason for this change?

**Review by evantk91 — COMMENTED** [review]
Had a few clarifications. But otherwise looking good.

File: example/src/hooks/useIterableApp.tsx [hunk 2]
```diff
@@ -45,10 +47,16 @@ interface IterableAppProps {
   loginInProgress?: boolean;
   /** Logs the user out */
   logout: () => void;
+  /** TODO: Ask @evantk91 or @Ayyanchira what this is for */
+  returnToInboxTrigger: boolean;
```
[the remark this answers is not in the document]
Comment (lposen) [reply example/src/hooks/useIterableApp.tsx:51]: Thanks!  I think I went over this with you in the documentation PR.

File: example/src/components/Inbox/Inbox.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {
+  IterableInbox,
+  type IterableInboxProps,
+} from '@iterable/react-native-sdk';
+
+import { useIterableApp } from '../../hooks';
+import { iterableInboxCustomization } from './Inbox.constants';
+
+export const Inbox = (props: IterableInboxProps) => {
+  const { returnToInboxTrigger } = useIterableApp();
```
[the remark this answers is not in the document]
Comment (lposen) [reply example/src/components/Inbox/Inbox.tsx:10]: Thanks!

File: src/IterableInbox.tsx [hunk 4]
```diff
@@ -30,15 +30,15 @@ const RNIterableAPI = NativeModules.RNIterableAPI;
 const RNEventEmitter = new NativeEventEmitter(RNIterableAPI);
 
 // TODO: Comment
-type InboxProps = {
+export interface IterableInboxProps {
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/IterableInbox.tsx:33]: interfaces are more easily extendable than types, so it's encourages to use them in third party libs as your users may need to extend them.  Hence the change from `type` to `interface`.

File: src/IterableInboxMessageList.tsx (hunk 1 above)
Comment (lposen) [reply src/IterableInboxMessageList.tsx:55]: Cleanliness/readability.

File: src/IterableInbox.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (evantk91) [reply src/IterableInbox.tsx:33]: Gotcha. Good change.

File: src/IterableInboxMessageList.tsx (hunk 1 above)
Comment (evantk91) [reply src/IterableInboxMessageList.tsx:55]: Gotcha. Can the same be done for deleteRow?

**Pull request merged**